### ISSUE TITLE
Accounting not working for switches that don't send NAS-IP-Address

### ISSUE THREAD

**Issue by lyubomirtraykov** [inverse-inc/packetfence#5722]
I am getting this errors on accounting with Juniper switches, because they don't send NAS-IP-Address on accounting.
`Jul 25 16:38:04 pf packetfence_httpd.aaa: httpd.aaa(1999) ERROR: [mac:00:25:64:bb:b2:06] WARNING ! Unknown switch(es) 08:81:f4:ab:43:c0 (pf::SwitchFactory::instantiate)`

`Jul 25 16:38:04 pf packetfence_httpd.aaa: httpd.aaa(1999) WARN: [mac:00:25:64:bb:b2:06] Use of uninitialized value $switch_ip in concatenation (.) or string at /usr/local/pf/lib/pf/radius.pm line 387. (pf::radius::accounting)`

`Jul 25 16:38:04 pf packetfence_httpd.aaa: httpd.aaa(1999) WARN: [mac:00:25:64:bb:b2:06] Can't instantiate switch (). This request will be failed. Are you sure your switches.conf is correct? (pf::radius::accounting)`

**Comment by julsemaan** [comment]
I believe we have a safety net in the FreeRADIUS unlang configuration to set the NAS-IP-Address to the RADIUS client IP when its not there.

I only see the config for this in the cluster virtual server though.

Question for @lyubomirtraykov: Are you running a cluster or standalone ?

Also, a workaround for the moment would be to define `08:81:f4:ab:43:c0` (value of Called-Station-Id) as a switch in PacketFence so that your packets are handled.

**Comment by lyubomirtraykov** [comment]
It is a standalone server.

**Comment by julsemaan** [comment]
I looked and I can't see the NAS-IP-Address detection logic for standalone but that may be me not looking properly.

@fdurand, is there a logic to fill the NAS-IP-Address if its missing for standalone servers

I found the logic in the packetfence-cluster virtual server here:
https://github.com/inverse-inc/packetfence/blob/e7a1d119d232e67ced4038f686c39b156fef044f/conf/radiusd/packetfence-cluster.example#L70

But can't find it in a virtual server that is called by a standalone server

**Comment by julsemaan** [comment]
@fdurand has confirmed that the logic doesn't apply to a standalone server.

We should wrap this in a policy and call it both in the packetfence-cluster and the packetfence virtual server

I'll do it and provide a patch you can test shortly

**Comment by lyubomirtraykov** [comment]
Ok thank you. I can confirm that the workaround works.

**Comment by julsemaan** [comment]
Hi,

I adjusted our FreeRADIUS configuration and pfacct so that it handles the case correctly in standalone.

In your case you'll want to use the pfacct with the fix

In order to do so:
```
systemctl stop packetfence-pfacct
curl https://support.inverse.ca/~jsemaan/pfacct-5722 > /usr/local/pf/sbin/pfacct
systemctl start packetfence-pfacct
```

And that will install the binary with the fix

Full fixes are in: #5723 but the new pfacct should be enough for you
